applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.cpp

```diff
@@ -1740,13 +1715,16 @@ int CrBeamElement3D2N::Check(const ProcessInfo& rCurrentProcessInfo) const
         }
 
         direction_vector_y = GetValue(LOCAL_AXIS_2);
+        const double norm_dir_y = MathUtils<double>::Norm(direction_vector_y);
 
-        KRATOS_ERROR_IF(MathUtils<double>::Norm(direction_vector_y)<numerical_limit)
-                << "Given LOCAL_AXIS_2 has length 0 for element " << Id() << std::endl;
+        KRATOS_ERROR_IF(norm_dir_y<numerical_limit) << "Given LOCAL_AXIS_2 has length 0 for element " << Id() << std::endl;
 
-        // a tollerance of 1e-3 allows for a rough deviation of 0.06 degrees from 90.0 degrees
-        KRATOS_ERROR_IF(std::abs(MathUtils<double>::Dot(direction_vector_x,direction_vector_y))>1e-3)
-                << "LOCAL_AXIS_1 is not perpendicular to LOCAL_AXIS_2 for element " << Id() << std::endl;
+        // a tolerance of 1e-3 allows for a rough deviation of 0.06 degrees from 90.0 degrees
+        const double dot_prod = MathUtils<double>::Dot(direction_vector_x, direction_vector_y);
+        if (std::abs(dot_prod)>1e-3) {

```

Comment:
Just one this, does this check of 1.0e-3 can be a property instead of hardcoded?

Reply:
hm not sure
afaik it should be pretty much zero (but for real cases at least a small tolerance is needed which is why we added this)

@KlausBSautter you know better, what happens if it is not fully perpendicular?

Reply:
then the inertia values do not represent what they should as far as I know. I would keep it hardcoded